pkg/controller/installation/windows_controller.go

```diff
@@ -0,0 +1,455 @@
+			typhaNodeTLS.TrustedBundle.AddCertificates(prometheusClientCert)
+		}
+	}
+
+	var component render.Component
+
+	if len(instance.Spec.ServiceCIDRs) == 0 {
+		r.status.SetDegraded(operatorv1.ResourceValidationError, "Calico for Windows requires at least one Kubernetes service CIDR to be configured in InstallationSpec.ServiceCIDRs", fmt.Errorf("InstallationSpec.ServiceCIDRs must be configured for Calico for Windows"), reqLogger)
+	}
+	if k8sapi.Endpoint.Host == "" || k8sapi.Endpoint.Port == "" {
+		r.status.SetDegraded(operatorv1.ResourceReadError, fmt.Sprintf("Services endpoint configmap '%s' does not have all required information for the Calico Windows daemonset configuration", render.K8sSvcEndpointConfigMapName), fmt.Errorf("Services endpoint configmap '%s' must be configured for Calico for Windows", render.K8sSvcEndpointConfigMapName), reqLogger)
+	}
+	if instance.Spec.CalicoNetwork != nil {
+		if v4pool := render.GetIPv4Pool(instance.Spec.CalicoNetwork.IPPools); v4pool != nil {
+			if v4pool.Encapsulation != operatorv1.EncapsulationVXLAN && v4pool.Encapsulation != operatorv1.EncapsulationNone {
+				r.status.SetDegraded(operatorv1.ResourceValidationError, "Encapsulation not supported by Calico for Windows", fmt.Errorf("IPv4 IPPool encapsulation %s is not supported by Calico for Windows", v4pool.Encapsulation), reqLogger)
+			}
+		}
+
+		// Retrieve DNS server addresses from DNS service ("kube-dns" in most providers, particular values on OpenShift and RKE2)
+		kubeDNSServiceName := types.NamespacedName{Name: "kube-dns", Namespace: "kube-system"}
+		if r.autoDetectedProvider == operator.ProviderOpenShift {
+			kubeDNSServiceName = types.NamespacedName{Name: "dns-default", Namespace: "openshift-dns"}
+		} else if r.autoDetectedProvider == operator.ProviderRKE2 {
+			kubeDNSServiceName = types.NamespacedName{Name: "rke2-coredns-rke2-coredns", Namespace: "kube-system"}
+		}
+
+		kubeDNSService := &corev1.Service{}
+		err = r.client.Get(ctx, kubeDNSServiceName, kubeDNSService)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				r.status.SetDegraded(operatorv1.ResourceNotFound, fmt.Sprintf("%s service not found", kubeDNSServiceName.Name), err, reqLogger)
+			} else {
+				r.status.SetDegraded(operatorv1.ResourceReadError, fmt.Sprintf("Error querying %s service", kubeDNSServiceName.Name), err, reqLogger)
+			}
+			return reconcile.Result{RequeueAfter: 10 * time.Second}, err
+		}
+		kubeDNSIPs := kubeDNSService.Spec.ClusterIPs
+
+		// felixConfiguration.Spec.VXLANVNI is defaulted by fillDefaults() in the core controller, so if it isn't set, requeue to wait until it is
+		if felixConfiguration.Spec.VXLANVNI == nil {
+			err = fmt.Errorf("VXLANVNI not specified in FelixConfigurationSpec")
+			r.status.SetDegraded(operatorv1.ResourceReadError, "Error reading VXLANVNI from FelixConfiguration", err, reqLogger)
+			return reconcile.Result{RequeueAfter: 10 * time.Second}, err
+		}
+
+		windowsCfg := render.WindowsConfiguration{
```

Comment:
Does this belong within the `if instance.Spec.CalicoNetwork != nil {` block?

Generally I'd expect we want to render the component regardless - the render package can decide to return no objects, though.